reframe/utility/__init__.py

```diff
@@ -24,6 +25,29 @@
 from . import typecheck as typ
 
 
+def get_hostname_cmd(autodetect_opts, logger=None):
```

Comment:
Since I don't think that this is needed to be called by the cli (see my comment there), it's better if this functionality is embedded directly in the auto-detection code inside the `_SiteConfig`.

Reply:
Also, there is generally no need to pass the logger here. `getlogger()` should always return a valid logger object, even if that it doesn't log anything.

Reply:
Since it's a `utility` function, I didn't want to import `reframe.core` function to get a logger.